controllers/elbv2/eventhandlers/endpoints.go

```diff
@@ -32,32 +33,32 @@ type enqueueRequestsForEndpointsEvent struct {
 }
 
 // Create is called in response to an create event - e.g. Pod Creation.
-func (h *enqueueRequestsForEndpointsEvent) Create(e event.CreateEvent, queue workqueue.RateLimitingInterface) {
+func (h *enqueueRequestsForEndpointsEvent) Create(ctx context.Context, e event.CreateEvent, queue workqueue.RateLimitingInterface) {
 	epNew := e.Object.(*corev1.Endpoints)
-	h.enqueueImpactedTargetGroupBindings(queue, epNew)
+	h.enqueueImpactedTargetGroupBindings(ctx, queue, epNew)
 }
 
 // Update is called in response to an update event -  e.g. Pod Updated.
-func (h *enqueueRequestsForEndpointsEvent) Update(e event.UpdateEvent, queue workqueue.RateLimitingInterface) {
+func (h *enqueueRequestsForEndpointsEvent) Update(ctx context.Context, e event.UpdateEvent, queue workqueue.RateLimitingInterface) {
 	epOld := e.ObjectOld.(*corev1.Endpoints)
 	epNew := e.ObjectNew.(*corev1.Endpoints)
 	if !equality.Semantic.DeepEqual(epOld.Subsets, epNew.Subsets) {
-		h.enqueueImpactedTargetGroupBindings(queue, epNew)
+		h.enqueueImpactedTargetGroupBindings(ctx, queue, epNew)
 	}
 }
 
 // Delete is called in response to a delete event - e.g. Pod Deleted.
-func (h *enqueueRequestsForEndpointsEvent) Delete(e event.DeleteEvent, queue workqueue.RateLimitingInterface) {
+func (h *enqueueRequestsForEndpointsEvent) Delete(ctx context.Context, e event.DeleteEvent, queue workqueue.RateLimitingInterface) {
 	epOld := e.Object.(*corev1.Endpoints)
-	h.enqueueImpactedTargetGroupBindings(queue, epOld)
+	h.enqueueImpactedTargetGroupBindings(ctx, queue, epOld)
 }
 
 // Generic is called in response to an event of an unknown type or a synthetic event triggered as a cron or
 // external trigger request - e.g. reconcile AutoScaling, or a WebHook.
-func (h *enqueueRequestsForEndpointsEvent) Generic(event.GenericEvent, workqueue.RateLimitingInterface) {
+func (h *enqueueRequestsForEndpointsEvent) Generic(context.Context, event.GenericEvent, workqueue.RateLimitingInterface) {
 }
 
-func (h *enqueueRequestsForEndpointsEvent) enqueueImpactedTargetGroupBindings(queue workqueue.RateLimitingInterface, ep *corev1.Endpoints) {
+func (h *enqueueRequestsForEndpointsEvent) enqueueImpactedTargetGroupBindings(ctx context.Context, queue workqueue.RateLimitingInterface, ep *corev1.Endpoints) {
 	tgbList := &elbv2api.TargetGroupBindingList{}
```

Comment:
why we need to pass `ctx` to `enqueueImpactedTargetGroupBindings()` but doesn't seem to use within this func? looks unnecessary here, unless I'm missing anything

Reply:
see comment on `controllers/ingress/eventhandlers/ingress_class_events.go`. Resolved.